plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/references/WildcardTypeReference.java

```diff
@@ -336,8 +336,10 @@ private String getAsString(Function<LightweightTypeReference, String> format) {
 		if (lowerBound != null) {
 			return "? super " + format.apply(lowerBound);
 		}
-		if (upperBounds != null && upperBounds.size() == 1 && upperBounds.get(0).isType(Object.class)) {
-			return "?";
+		if (format == SimpleNameFunction.INSTANCE) {
```

Comment:
Why not change the signature of the private method and add a boolean flag to make the behavior more explicit?